repository-azure/src/integration-test/java/io/aiven/elasticsearch/repositories/azure/AzureRepositoryPluginIT.java

```diff
@@ -0,0 +1,149 @@
+                                            .setString(
+                                                    AzureStorageSettings.AZURE_ACCOUNT_KEY.getKey(),
+                                                    azureAccountKey)
+                                            .setFile(
+                                                    AzureStorageSettings.PUBLIC_KEY_FILE.getKey(),
+                                                    Files.newInputStream(publicKeyPem))
+                                            .setFile(
+                                                    AzureStorageSettings.PRIVATE_KEY_FILE.getKey(),
+                                                    Files.newInputStream(privateKeyPem))
+                                    ));
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        final var blobServiceClient =
+                new BlobServiceClientBuilder()
+                        .connectionString(
+                                String.format(
+                                        AzureStorageSettings.AZURE_CONNECTION_STRING_TEMPLATE,
+                                        azureAccount,
+                                        azureAccountKey)
+                        ).buildClient();
+        blobContainerClient = blobServiceClient.getBlobContainerClient(containerName);
+    }
+
+    @AfterAll
+    static void tearDown() throws IOException {
+        try {
+            shutdownCluster();
+        } finally {
+            listOfFileForBackupFolder(BASE_PATH)
+                    .forEach(f -> blobContainerClient.getBlobClient(f).delete());
+        }
+    }
+
+    @Override
+    public String registerRepositoryJson() {
+        return "{ \"type\": \""
+                + AzureRepositoryPlugin.REPOSITORY_TYPE + "\", "
+                + "\"settings\": { \"container_name\": \""
+                + containerName + "\", \"base_path\": \"" + BASE_PATH + "\" } "
+                + "}";
+    }
+
+    @Override
+    protected void assertRegisterRepository(final String responseContent) {
+        assertEquals(
+                "{backup={settings={container_name=" + containerName + ", base_path=" + BASE_PATH + "}, "
+                        + "type=" + AzureRepositoryPlugin.REPOSITORY_TYPE + "}}", responseContent
+        );
+    }
+
+    @Override
+    public void assertCreatedSnapshot() {
+        final var metadataFile = BASE_PATH + "/" + RepositoryStorageIOProvider.REPOSITORY_METADATA_FILE_NAME;
+        final var metadataBlob = blobContainerClient
+                .getBlobClient(metadataFile)
+                .exists();
+        assertTrue(metadataBlob);
+
+        final var listOfFilesIterator = listOfFileForBackupFolder(BASE_PATH);
+        assertNotNull(listOfFilesIterator);
+        assertTrue(countElementsInBackup(listOfFilesIterator) > 0);
+    }
+
+    @Override
+    public void assertDeletionOfRepository() {
```

Comment:
Sorry, don't get what happened in this test. When deletion is happening??

Reply:
When deletion is happening ES doesn't delete the repo it deletes only indices snapshots, all other files remains untouched